core/cat/routes/settings.py

```diff
@@ -102,3 +108,41 @@ def delete_setting(
     crud.delete_setting_by_id(settingId)
 
     return {"deleted": settingId}
+
+
+@router.post("/factory-reset")
+def factory_reset(
+    request: Request,
+    stray=Depends(HTTPAuth(AuthResource.SETTINGS, AuthPermission.WRITE)),
+):
+    """Delete Cat's memory and metadata.json file (settings db).
+    A 'soft restart' is performed recreating the Cat instance and the metadata.json file.
+    Plugins are not affected but need to be switched on.
+    """
+
+    metadata_file = get_env("CCAT_METADATA_FILE")
+    metadata_content = {}
+
+    try:
+        if os.path.exists(metadata_file):
+            with open(metadata_file, 'r') as file:
+                metadata_content = json.load(file)
+
+            os.remove(metadata_file)
+
+        vector_memory: VectorMemory = stray.memory.vectors
+        deleted_memories = utils.delete_collections(vector_memory)
+
+        utils.singleton.instances.clear()
+        request.app.state.strays.clear()
+
+        request.app.state.ccat = CheshireCat()
```

Comment:
does it work with a singleton, i.e. a class which constructor is triggered the first time only?